display/ssd1306.h

```diff
@@ -225,15 +229,37 @@ class SSD1306 : public I2CDevice, Looper, StateMachine, SaberBase, private Audio
 	}
 	if (eof_) {
 	  // STDOUT << "EOF " << frame_count_ << "\n";
-	  screen_ = SCREEN_PLI;
-	  if (frame_count_ == 1) return 5000;
-	  return FillFrameBuffer();
+    if (!SaberBase::IsOn()) {
+	   screen_ = SCREEN_PLI;
+	    if (frame_count_ == 1) return font_config.ProffieOSImageDuration;
+	    return FillFrameBuffer();
+    }
 	}
 	frame_count_++;
-	if (looped_frames_ > 1 && millis() - loop_start_ > 5000) {
-	  STDOUT << "END OF LOOP\n";
-	  screen_ = SCREEN_PLI;
-	}
+  if (SaberBase::IsOn()) {
+    if (looped_frames_ == 1 && frame_count_ == 1) {
+      return font_config.ProffieOSImageDuration;
+    } else if (looped_frames_ == 1 && frame_count_ > 1) {
+      if (!SaberBase::Lockup()) {
+        if (!imgon_) {
+          ShowFile(&IMG_on);
+        } else {
+          screen_ = SCREEN_PLI;
+        }
+      } else {
+        ShowFile(&IMG_lock);
```

Comment:
What happens if there are no IMG_lock file(s)?
We should still play IMG_on in that case, right?
(And if there is no IMG_on, we should go back to PLI?)

Reply:
Added a screen_ = SCREEN_PLI before invoking ShowFile, which covers off a missing on.bmp.  Missing lock.bmp is already handled by the check within ShowFile for the effect existing.